gax-httpjson/src/main/java/com/google/api/gax/httpjson/HttpJsonClientCall.java

```diff
@@ -0,0 +1,158 @@
+     * processed by the server. No further receiving will occur and no further notifications will be
+     * made.
+     *
+     * <p>This method should not throw. If this method throws, there is no way to be notified of the
+     * exception. Implementations should therefore be careful of exceptions which can accidentally
+     * leak resources.
+     *
+     * @param statusCode the HTTP status code representing the result of the remote call
+     * @param trailers metadata provided at call completion
+     */
+    public void onClose(int statusCode, HttpJsonMetadata trailers) {}
+  }
+
+  /**
+   * Start a call, using {@code responseListener} for processing response messages.
+   *
+   * <p>It must be called prior to any other method on this class, except for {@link #cancel} which
+   * may be called at any time.
+   *
+   * @param responseListener receives response messages
+   * @param requestHeaders which can contain extra call metadata, e.g. authentication credentials.
+   */
+  public abstract void start(Listener<ResponseT> responseListener, HttpJsonMetadata requestHeaders);
+
+  /**
+   * Requests up to the given number of messages from the call to be delivered to {@link
+   * HttpJsonClientCall.Listener#onMessage(Object)}. No additional messages will be delivered.
+   *
+   * <p>Message delivery is guaranteed to be sequential in the order received. In addition, the
+   * listener methods will not be accessed concurrently. While it is not guaranteed that the same
+   * thread will always be used, it is guaranteed that only a single thread will access the listener
+   * at a time.
+   *
+   * <p>If called multiple times, the number of messages able to delivered will be the sum of the
+   * calls.
+   *
+   * <p>This method is safe to call from multiple threads without external synchronization.
+   *
+   * @param numMessages the requested number of messages to be delivered to the listener. Must be
+   *     non-negative.
+   */
+  public abstract void request(int numMessages);
+
+  /**
+   * Prevent any further processing for this {@code HttpJsonClientCall}. No further messages may be
+   * sent or will be received. The server is not informed of cancellations. Cancellation is
+   * permitted even if previously {@link #halfClose}d. Cancelling an already {@code cancel()}ed
+   * {@code ClientCall} has no effect.
+   *
+   * <p>No other methods on this class can be called after this method has been called.
+   *
+   * @param message if not {@code null}, will appear as the description of the CANCELLED status
+   * @param cause if not {@code null}, will appear as the cause of the CANCELLED status
+   */
+  public abstract void cancel(@Nullable String message, @Nullable Throwable cause);
+
+  /**
+   * Close the call for request message sending. Incoming response messages are unaffected. This
+   * should be called when no more messages will be sent from the client.
+   */
+  public void halfClose() {}
```

Comment:
Why isn't this `abstract`? It is in `io.grpc.CllientCall`.

Reply:
made abstract